engine/runner/src/main/java/org/enso/runner/EnsoLibraryFeature.java

```diff
@@ -90,46 +89,31 @@ public void beforeAnalysis(BeforeAnalysisAccess access) {
               }
             }
           }
+          if (pkg.nativeLibraryDir().exists()) {
+            var nativeLibs =
+                NativeLibraryFinder.listAllNativeLibraries(pkg, FileSystem$.MODULE$.defaultFs());
+            for (var nativeLib : nativeLibs) {
+              assert nativeLib.exists() && nativeLib.isFile();
+              var dir = nativeLib.toPath().getParent().toFile();
+              assert dir.exists() && dir.isDirectory();
+              nativeLibPaths.add(dir.getAbsolutePath());
+              var current = System.getProperty("java.library.path");
+              RuntimeSystemProperties.register(
+                  "java.library.path", current + File.pathSeparator + dir.getAbsolutePath());
+            }
+          }
         }
       }
     } catch (Exception ex) {
-      ex.printStackTrace();
+      ex.printStackTrace(System.err);
       throw new IllegalStateException(ex);
     }
     System.err.println("Summary for polyglot import java:");
     for (var className : classes) {
       System.err.println("  " + className);
     }
     System.err.println("Registered " + classes.size() + " classes for reflection");
-  }
-
-  private static void registerOpenCV(ClassLoader cl) throws ReflectiveOperationException {
-    var moduleOpenCV = cl.getUnnamedModule();
-    var currentOS = System.getProperty("os.name").toUpperCase().replaceAll(" .*$", "");
-
-    var libOpenCV =
-        switch (currentOS) {
-          case "LINUX" -> "nu/pattern/opencv/linux/x86_64/libopencv_java470.so";
-          case "WINDOWS" -> "nu/pattern/opencv/windows/x86_64/opencv_java470.dll";
-          case "MAC" -> {
-            var arch = System.getProperty("os.arch").toUpperCase();
-            yield switch (arch) {
-              case "X86_64" -> "nu/pattern/opencv/osx/x86_64/libopencv_java470.dylib";
-              case "AARCH64" -> "nu/pattern/opencv/osx/ARMv8/libopencv_java470.dylib";
-              default -> null;
-            };
-          }
-          default -> null;
-        };
-
-    if (libOpenCV != null) {
-      var verify = cl.getResource(libOpenCV);
-      if (verify == null) {
-        throw new IllegalStateException("Cannot find " + libOpenCV + " resource in " + cl);
-      }
-      RuntimeResourceAccess.addResource(moduleOpenCV, libOpenCV);
```

Comment:
Please note that the `RuntimeResourcesAccess` is different. It really puts the library into the _native executable_ in build time. However we don't want to do it.

- this whole `registerOpenCV` section added by #11807 should be deleted

Reply:
As of a1521cd23edbb62fecb332d53931ddc93c72d169 the whole section is already deleted.